### PR TITLE
Clarify sudoku solver instructions

### LINKED ISSUE #40413: Quality Assurance Projects - Sudoku Solver clearer instructions on logic
**Describe your problem and how to reproduce it:**
There are two major questions I had while I was doing this project, and it seems like other people also had similar issues.

1.) (There's a similar issue to this already [here](https://github.com/freeCodeCamp/freeCodeCamp/issues/40359), but this question deals with conflicting row/column/region.)

Q: Do we need to indicate if the value at a coordinate is conflicting with the final solution or with what is currently on the board? 

If something could indicate that you only need to figure out checkRow/Column/RegionPlacement in `sudoku-solver.js` with what is currently on the board rather than checking with the final solution, that would help immensely.

2.) Q: For the `solve()` method in `sudoku-solver.js`, does it need to check against solution logic provided elsewhere in the backend only on the test inputs, or do we need to provide the solve logic ourselves for ANY given puzzleString input? 

I received several conflicting answers when I asked on Discord, but I believe the right answer is that I need to provide the logic myself? Making this clearer on the /learn page by saying that we need to return the solved string with our own logic in `solve()` would be helpful.

These are probably not the clearest answers, but these were sufficient in helping me answer my questions that I had about this.

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/quality-assurance/quality-assurance-projects/sudoku-solver

### PR DESCRIPTION
**Pull request by vicky11z**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #40413

Tested locally, here is what it looks like:
Screen Shot 2021-01-01 at 10 10 22 PM

### CODE REVIEW

File: curriculum/challenges/english/06-quality-assurance/quality-assurance-projects/sudoku-solver.md [hunk 1]
```diff
@@ -17,6 +17,9 @@ When you are done, make sure a working demo of your project is hosted somewhere
 # --instructions--
 
 -   All puzzle logic can go into `/controllers/sudoku-solver.js`
+    - The `validate` function should take a given puzzle string and check it to see if it has 81 valid characters for the input.
+    - The `check` functions should be validating against the *current* state of the board.
+    - The `solve` function should handle solving any given valid puzzle string, not just the test inputs and solutions. You are expected to write out the logic to solve this.
 -   All routing logic can go into `/routes/api.js`
```
Comment (nhcarrigan) [inline curriculum/challenges/english/06-quality-assurance/quality-assurance-projects/sudoku-solver.md:23]:
```suggestion
-   All puzzle logic can go into `/controllers/sudoku-solver.js`
  - The `validate` function should take a given puzzle string and check it to see if it has 81 valid characters for the input.
  - The `check` functions should be validating against the *current* state of the board.
  - The `solve` function should handle solving any given valid puzzle string, not just the test inputs and solutions. You are expected to write out the logic to solve this.
-   All routing logic can go into `/routes/api.js`
```
This should correct the linting issue. 🙂

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Comment by RandellDawson** [comment]
I honestly had not noticed the new parser makes the lists indented by 4.  However, if we change the entire `Instructions`section to the following, it looks as intended and the linter no longer complains.
```markdown
- All puzzle logic can go into `/controllers/sudoku-solver.js`
- The `validate` function should take a given puzzle string and check it to see if it has 81 valid characters for the input.
  - The `check` functions should be validating against the *current* state of the board.
  - The `solve` function should handle solving any given valid puzzle string, not just the test inputs and solutions. You are expected to write out the logic to solve this.
- All routing logic can go into `/routes/api.js`
- See the `puzzle-strings.js` file in `/controllers` for some sample puzzles your application should solve
- To run the challenge tests on this page, set `NODE_ENV` to `test` without quotes in the `.env` file
- To run the tests in the console, use the command `npm run test`. To open the Repl.it console, press Ctrl+Shift+P (Cmd if on a Mac) and type "open shell"
- ```

**Comment by vicky11z** [comment]
Thanks for taking a look! I took @RandellDawson 's recommendation since when I just tried to decrease the spaces down to 2, the instructions I added were no longer at the correct indentation level.

**Review by RandellDawson — APPROVED** [review]
LGTM

@nhcarrigan What do you think?

**Review by naomi-lgbt — APPROVED** [review]
LGTM~! 🥳 

Thank you for your pull request! We look forward to additional contributions!

Please note that we are currently in a code freeze and are not merging PRs to curriculum lessons for the time being. 🙂

**Pull request merged**